flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/ResultPartitionManager.java

```diff
@@ -40,9 +44,18 @@
 
 	private static final Logger LOG = LoggerFactory.getLogger(ResultPartitionManager.class);
 
+	/**
+	 * Table of running/finished ResultPartitions with corresponding IDs
+	 */
 	public final Table<ExecutionAttemptID, IntermediateResultPartitionID, ResultPartition>
 			registeredPartitions = HashBasedTable.create();
 
+	/**
+	 * Cached ResultPartitions which are used to resume/recover from
```

Comment:
There are two unused classes LRUCache and LRUCacheMap, which can be removed. I think this solution is fine. :)